### PR TITLE
Jun2 resize ssr with master

### PR DESCRIPTION
**Pull request by guysopher**

### CODE REVIEW

File: packages/gallery/src/common/utils/index.js [hunk 1]
```diff
@@ -283,8 +282,9 @@ class Utils {
     });
   }
 
+  //TODO : Replace with isPrerender mode
```
Comment (guysopher) [inline packages/gallery/src/common/utils/index.js:285]: should we do the todo?
(no ssr references in the core)

File: packages/gallery/src/components/gallery/index.js [hunk 2]
```diff
@@ -7,6 +7,7 @@ import dimensionsHelper from '../helpers/dimensionsHelper';
 import defaultStyles from '../../common/defaultStyles';
 import LAZY_LOAD from '../../common/constants/lazyLoad';
 import utils from '../../common/utils';
+import Blueprints from '../blueprints/Blueprints'
```
Comment (guysopher) [inline packages/gallery/src/components/gallery/index.js:10]: this should not be here

File: packages/gallery/src/components/gallery/proGallery/slideshowView.js [hunk 3]
```diff
@@ -295,7 +295,7 @@ class SlideshowView extends GalleryComponent {
   }
 
   autoScrollToNextItem = () => {
-    if (!isEditMode() && (isGalleryInViewport(this.props.container) || isPreviewMode())) {
+    if (!isEditMode() && (isGalleryInViewport(this.props.container))) {
```
Comment (guysopher) [inline packages/gallery/src/components/gallery/proGallery/slideshowView.js:298]: is this safe to remove?
What happens in preview?

File: packages/gallery/src/components/item/imageItem.js [hunk 4]
```diff
@@ -67,58 +95,59 @@ export default class ImageItem extends GalleryComponent {
...
+              {...preloadProps}
+            />
+            break;
+        }
+
+        imagesComponents.push(preload);
+      }
 
-      return [preload, highres]
+      const shouldRenderHighResImages = !isPrerenderMode() && !utils.isSSR();
```
Comment (guysopher) [inline packages/gallery/src/components/item/imageItem.js:133]: why are we using isSSR?
We should use componentDidMount to allow highres images

File: packages/gallery/src/components/styles/animations.scss [hunk 5]
```diff
@@ -0,0 +1,5 @@
+:global {
+    .animation-slide {
+        transition: all 0.4s ease
```
Comment (guysopher) [inline packages/gallery/src/components/styles/animations.scss:3]: this is very bad css.
Use the actual properties you need to animate - never use `all`

File: packages/ssr-simulator/src/components/gallery/gallery.js [hunk 6]
```diff
@@ -3,28 +3,100 @@ import { ProGallery } from 'pro-gallery';
...
-    const items = utils.mixAndSlice(testItems, 50, styles.seed || 1);
+    this.handleResize = this.handleResize.bind(this);
+
+    this.state = {
+      knownContainer: container !== UNKNOWN_CONTAINER,
+      container: container
+    };
+  }
+
+  initContainer(props) {
```
Comment (guysopher) [inline packages/ssr-simulator/src/components/gallery/gallery.js:27]: it is actually - `getContainerFromUrl`

File: packages/gallery/src/common/utils/index.js (hunk 1 above)
Comment (aviranhuga) [reply packages/gallery/src/common/utils/index.js:285]: after santa-wrapper will support PRERENDER mode

File: packages/gallery/src/components/gallery/index.js (hunk 2 above)
Comment (aviranhuga) [reply packages/gallery/src/components/gallery/index.js:10]: deleted

File: packages/gallery/src/components/item/imageItem.js (hunk 4 above)
Comment (aviranhuga) [reply packages/gallery/src/components/item/imageItem.js:133]: backward compatibility, we will remove it when Santa will use prerender mode

**Pull request closed without merging**